OneCommunityGlobal/HGNRest
# Strallia/Faye: Data for Volunteer Status Pie Chart on Total Org Summary Page

# Description
This PR retrieves the data required by the Volunteer Status pie chart on the Total Org Summary Page. Specifically, it formats the API response and retrieves comparison data for the number of total volunteers.

## Related PRS (if any):
This PR is built off the backend PR https://github.com/OneCommunityGlobal/HGNRest/pull/1062.

## Main changes explained:
- Update overviewReportHelper.js with aggregation pipelines to gather comparison data for the chart

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. use Postman or curl commands to get an authorization token at http://localhost:4500/api/login using a JSON body with your developer account credentials:
```
{
    "email": "<YOUR EMAIL>"
    "password": "<YOUR PASSWORD>"
}
```
4. include the token in a request to localhost:4500/api/reports/volunteerstats?startDate=2024-05-26&endDate=2024-06-01&comparisonStartDate=2024-05-19&comparisonEndDate=2024-05-25
5. verify these highlighted fields are returned in the response:

<img width="955" alt="Screenshot 2024-12-02 at 10 28 07 AM" src="https://github.com/user-attachments/assets/a07bedaa-0e9c-487d-bf2d-613a67f5b949">

6. verify that a valid response is still returned when the optional parameters comparisonStartDate and comparisonEndDate are not included.
